pennylane/ops/qubit.py

```diff
@@ -192,12 +215,16 @@ class S(Operation):
     num_wires = 1
     par_domain = None
 
```

Comment:
How come sometimes `eigvals` class attribute is not provided?

Reply:
I suppose in this case it just falls back to the default implementation `np.linalg.eigvalsh(self.matrix)`? But hardcoding it might reduce the overhead of constantly calling `eigvalsh`.

Reply:
The `eigvals` class attirbute is only used for subclasses of `Observable`

Reply:
The point is that `eigvals` are not used besides for `DiagonalOperation` instances. So I didn't want to spend too much time figuring out the eigenvalue formula of `Rot` and the like if nobody uses it anyways.